resolvers.py

```diff
@@ -0,0 +1,1979 @@
+            qs['value'] = self.merge_allocations_count(qs['value'])
+
+            if qs["account_name"] == 'Total':
+                qs_dummy = qs["value"]
+                for item in qs_dummy:
+                    string = item["period_name"]+" "+item["week_name"]+" "+item["start_date"] + \
+                        " "+item["end_date"]+" " + \
+                        item["record_type"]+" "+item["category"]
+                    if hashmap.get(string) is not None:
+                        hashmap[string] = int(
+                            item["allocations"]) + int(hashmap.get(string))
+                    else:
+                        hashmap[string] = int(item["allocations"])
+
+        return hashmap
+
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+
+        billability = kwargs.get("billability")
+
+        # get current period
+        current_period = models.Periods.objects.filter(
+            period_tag="current_period").values()[0]
+
+        current_period_name = current_period["period_name"]
+        current_period_start_date = current_period["period_start_date"]
+
+        qs = core_models.AccountWiseHeadcountSnapshot.objects.all()
+
+        qs_result = {}
+
+        # going over all the data of backup
+        for val in qs.values():
+            category = "Previous Period"
+
+            if current_period_name == val["period_name"]:
+
+                # to check if data is of current period
+                if val["start_date"] < current_period_start_date:
+                    continue
+                category = "Current Period"
+
+            snapshot = json.loads(val["snapshot_data"])
+
+            snapshot_data = self.snapshot_helper(billability, snapshot)
+
+            # going over snapshot data of every row in backup table.
+            for snap_val in snapshot_data:
+                # key to seperate different account_name and project_name type
+                key = snap_val["account_name"] + \
+                    "_" + snap_val["project_name"]
+
+                data_value = {
+                    "period_name": val["period_name"],
+                    "week_name": val["week_name"],
+                    "start_date": str(val["start_date"]),
+                    "end_date": str(val["end_date"]),
+                    "record_type": val["record_type"],
+                    "category": category,
+                    "account_name": snap_val["account_name"],
+                    "project_name": snap_val["project_name"],
+                    "billability": snap_val["billability"],
+                    "allocations": snap_val["allocations"]
+                }
+
+                if key not in qs_result:
+                    qs_result[key] = {"account_name": snap_val["account_name"],
+                                      "project_name": snap_val["project_name"], "value": []}
+
+                qs_result[key]["value"].append(data_value)
+
+        if len(qs_result) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)